scripts/global/player_inventory.gd

```diff
@@ -0,0 +1,142 @@
+extends Node
+
+signal active_item_updated
+signal inventory_active_item_updated
+
+# Constants defining the number of inventory and hotbar slots
+const NUM_INVENTORY_SLOTS = 20
+const NUM_HOTBAR_SLOTS = 8
+
+# Preloaded classes for Item and Slot
+const ItemClass = preload("res://scripts/ui/inventory/item.gd")
+const SlotClass = preload("res://scripts/ui/inventory/slot.gd")
+
+# Player Inventory Dictionary
+var inventory = {
```

Comment:
Another thing I have to object to. For one, you're using a dictionary with normal keys, meaning it's just an array but as dictionary. Unless your array is massive, this is rarely a good option for a variety of reasons (re: ptsd from dictionary duplication bugs)